### ISSUE TITLE
Admin sends `finishTime` too late 

### ISSUE THREAD

**Issue by yatsunastya** [skalenetwork/skaled#1625]
**Steps to Reproduce/Description**
1. Stop skaled on node A (`20`)
2. Run initExit and node Exit on node B
3. Wait for rotation delay
4. Start skaled on node A
5. Check skaled logs 

**Actual result:**
Skaled caught up all the blocks before finish TS (including the last one - when it has to exit) with old keys, and only after that admin sent `finishTime`. So skaled is waiting for the next block to mine and exit. 

```
[2023-08-18 17:12:24.427] [20:main] [error] 1897679:Could not BLS verify signature:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX0356260897930171323640843822186387979895:XXXXXXXXX759078738947693673371369133472091657581470309436154656373376581247:XXXXXXXXXXXXXXXXX1176122673834023157557573234606763785976268439975220814527:0:KEY:XXXXXXXXXXXXXXXXXXXXXXXXXXX2558485904513952254059510181435197668771355:HASH:XXXXXXXXXXXXXXXXXXXXXX3419a8bc58fae537cafb1de4f57768a264cbe5ec
[2023-08-18 17:12:24.435] [20:main] [error] 1897679:Successfully deserialized 0 blocks, got exception on block 1897680
[2023-08-18 17:12:24.435] [20:catchup] [error] 1897679:Catchupc step 3: can not read missing blocks
[2023-08-18 17:12:24.435] [20:main] [error] 1897679:!Exception: CatchupClientAgent:Catchupc step 3: can not read missing blocks
[2023-08-18 17:12:24.436] [20:main] [error] 1897679: !Caused by: CatchupClientAgent:Could not process block list
[2023-08-18 17:12:24.436] [20:main] [error] 1897679:  !Caused by: vector::_M_range_check: __n (which is 0) >= this->size() (which is 0)
```

**Expected behavior**
Admin sends `finishTime` at the start of skaled, before catch up. 
It's necessary to investigate if it's possible to fix it fast or to postpone to 2.3 release  

**Version:** 
admin: 2.5.1-develop.2
skaled: 3.17.0-beta.7

**Logs:**
[skaled_catch_up.txt](https://github.com/skalenetwork/skaled/files/12382574/skaled_catch_up.txt)
[skaled_staying_node_before_exit.txt](https://github.com/skalenetwork/skaled/files/12395917/skaled_staying_node_before_exit.txt)
[skaled_staying_node_after_exit.txt](https://github.com/skalenetwork/skaled/files/12395919/skaled_staying_node_after_exit.txt)

```[tasklist]
### Depends on
- [ ] https://github.com/skalenetwork/internal-support/issues/450
```

**Comment by PolinaKiporenko** [comment]
was found during 2.2 release testing

**Comment by PolinaKiporenko** [comment]
related to https://github.com/skalenetwork/internal-support/issues/450

**Comment by PolinaKiporenko** [comment]
should be solved after https://github.com/skalenetwork/internal-support/issues/450

**Comment by badrogger** [comment]
Fixed in v2.6.x versions

**Comment by DmytroNazarenko** [comment]
skale-admin: 2.6.0-beta.0
skaled: 3.18.0-beta.0

**Comment by yatsunastya** [comment]
Checked  ✅
Admin: 2.6.0-beta.0

After turning on skaled (after rotation delay) it catches up all blocks before `Exit TS` and finishes its work (based on TS in `rotation.json` file which is already in the past.) After that admin recreates skaled container (and ima). Skaled catches up all blocks after rotation delay till current timestamp and starts mining new blocks.
